### PR TITLE
Added short Python Solution for usaco-894 Grass Planting

### PR DESCRIPTION
**Pull request by RedBlueBird**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-894.mdx [hunk 1]
```diff
@@ -117,6 +117,29 @@ public class GrassPlanting {
...
+
+```py
+fin = open("planting.in", 'r')
+fout = open('planting.out', 'w')
+inputs = [i for i in fin.read().splitlines()]
+
+n = int(inputs[0])
+deg = [1 for i in range(n+1)]
+maxDeg = 0
+
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:130]: use snake_case

File: solutions/usaco-894.mdx [hunk 2]
```diff
@@ -117,6 +117,29 @@ public class GrassPlanting {
...
+# Tracking max degree
+for i in inputs[1:]:
+	u,v = [int(j) for j in i.split(" ")]
+	deg[u] += 1 
+	deg[v] += 1 
+	maxDeg = max(maxDeg, deg[u], deg[v]);
+
+print(maxDeg, file=fout)
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:138]: `fout` is redundant, just directly put it in here

File: solutions/usaco-894.mdx [hunk 3]
```diff
@@ -117,6 +117,29 @@ public class GrassPlanting {
...
+fout = open('planting.out', 'w')
+inputs = [i for i in fin.read().splitlines()]
+
+n = int(inputs[0])
+deg = [1 for i in range(n+1)]
+maxDeg = 0
+
+# Tracking max degree
+for i in inputs[1:]:
+	u,v = [int(j) for j in i.split(" ")]
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:133]: spaces and stuff

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-894.mdx (hunk 2 above)
Comment (RedBlueBird) [reply solutions/usaco-894.mdx]: like
`print(max_deg)`
`print(max_deg, fout)`
`print(max_deg, file)`?
I tried those 3 and it doesn't seem to work

Comment (SansPapyrus683) [reply solutions/usaco-894.mdx]: `print(max_deg, open("filename.out", "w"))`

**Comment by SansPapyrus683** [comment]
jeff if you can could you also do a python for sol 2?

File: solutions/usaco-894.mdx [hunk 4]
```diff
@@ -117,6 +117,28 @@ public class GrassPlanting {
 
 </JavaSection>
 
+<PySection>
+
+```py
+fin = open('planting.in', 'r')
+inputs = [i for i in fin.read().splitlines()]
```
Comment (ryanchou-dev) [inline solutions/usaco-894.mdx:124]: wait can't this just be replaced with
```
fin.read().splitlines()
```
or
```
fin.readlines()
```

**Review by ryanchou-dev — APPROVED** [review]
great

File: solutions/usaco-894.mdx (hunk 4 above)
Comment (nwatx) [reply solutions/usaco-894.mdx:124]: ^

**Comment by ryanchou-dev** [comment]
stop.

https://github.com/cpinitiative/usaco-guide/pull/2641#discussion_r835822161
@RedBlueBird address this ^^

**Comment by RedBlueBird** [comment]
> jeff if you can could you also do a python for sol 2?

Ok done

File: solutions/usaco-894.mdx [hunk 5]
```diff
@@ -146,41 +169,169 @@ void setIO(string name = ""){ // FastIO see General -> Fast Input and Output
...
+		}
+
+		int ret = 0;
+
+		for(int i = 1; i <= n; i++) ret = Math.max(ret, deg[i]);
+		io.println(ret+1);
+		io.close();
+	}
+
+	private static class Kattio extends PrintWriter {
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:215]: you can just use `//CodeSnip{Kattio}` i think

File: solutions/usaco-894.mdx [hunk 6]
```diff
@@ -146,41 +169,169 @@ void setIO(string name = ""){ // FastIO see General -> Fast Input and Output
...
+```
+
+</JavaSection>
+
+<PySection>
+
+```py
+fin = open("planting.in", 'r')
+fout = open('planting.out', 'w')
+inputs = fin.read().splitlines()
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:256]: use only single quotes or double quotes, don't do both

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-894.mdx (hunk 5 above)
Comment (RedBlueBird) [reply solutions/usaco-894.mdx:215]: how does that work

Comment (SansPapyrus683) [reply solutions/usaco-894.mdx:215]: just replace the entire class with 
```java
	//CodeSnip{Kattio}
```

File: solutions/usaco-894.mdx [hunk 7]
```diff
@@ -81,42 +81,54 @@ public class GrassPlanting {
...
-		// USACO-style file input
-		public Kattio(String problemName) throws IOException {
-			super(new FileWriter(problemName+".out"));
-			r = new BufferedReader(new FileReader(problemName+".in"));
-		}
+	// USACO-style file input
+	public Kattio(String problemName) throws IOException {
+		super(new FileWriter(problemName+".out"));
+		r = new BufferedReader(new FileReader(problemName+".in"));
+	}
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:90]: why's this stuff still here

File: solutions/usaco-894.mdx [hunk 8]
```diff
@@ -132,55 +144,52 @@ valid assignment of grass types in the $\texttt{ans}$ array.
...
-			ans[i] = g;
-			dfs(i, self);
-			g++;
-		}
-	}
+    int g = 1;
+    for (int i : adjList[self]) {
+        if (i != parent) {
+            while (g == ans[self] || g == ans[parent])
+                g++;
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:167]: braces

File: solutions/usaco-894.mdx [hunk 9]
```diff
@@ -132,55 +144,52 @@ valid assignment of grass types in the $\texttt{ans}$ array.
...
+        adjList[u].push_back(v);
+        adjList[v].push_back(u);
+    }
+    ans[1] = 1;
+    dfs(1, 1);
+    int maxAns = 0;
+    for (int curAns : ans) {
+        maxAns = max(maxAns, curAns);
+    }
+    cout << maxAns << "\n";
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:192]: use endl

File: solutions/usaco-894.mdx [hunk 10]
```diff
@@ -132,55 +144,52 @@ valid assignment of grass types in the $\texttt{ans}$ array.
...
+    int u, v;
+    for (int i = 1; i < n; i++){
+        cin >> u >> v;
+        adjList[u].push_back(v);
+        adjList[v].push_back(u);
+    }
+    ans[1] = 1;
+    dfs(1, 1);
+    int maxAns = 0;
+    for (int curAns : ans) {
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:189]: snake_case for c++

File: solutions/usaco-894.mdx [hunk 11]
```diff
@@ -237,4 +246,40 @@ public class Main {
 
 </JavaSection>
 
-</LanguageSection>
+<PySection>
+
+```py
+import sys 
+fin = open("planting.in", 'r')
+fout = open('planting.out', 'w')
```
Comment (SansPapyrus683) [inline solutions/usaco-894.mdx:254]: just put this directly in that print statement where you use it

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-894.mdx [hunk 12]
```diff
@@ -132,55 +144,52 @@ valid assignment of grass types in the $\texttt{ans}$ array.
...
-	}
-	cout << maxAns << "\n";
+    setIO("planting");
+
+    int n;
+    cin >> n;
+    int u, v;
+    for (int i = 1; i < n; i++){
+        cin >> u >> v;
+        adjList[u].push_back(v);
```
Comment (nwatx) [inline solutions/usaco-894.mdx:183]: should use tabs?

File: solutions/usaco-894.mdx [hunk 13]
```diff
@@ -81,42 +81,48 @@ public class GrassPlanting {
...
-		public double nextDouble() { return Double.parseDouble(next()); }
-		public long nextLong() { return Long.parseLong(next()); }
+		} catch (Exception e) {}
+		return null;
 	}
+
+	public int nextInt() { return Integer.parseInt(next()); }
+	public double nextDouble() { return Double.parseDouble(next()); }
+	public long nextLong() { return Long.parseLong(next()); }
 }
```
Comment (TruaShamu) [inline solutions/usaco-894.mdx:99]:
```// returns null if no more input
	public String next() {
		try {
			while (st == null || !st.hasMoreTokens())
				st = new StringTokenizer(r.readLine());
				return st.nextToken();
		} catch (Exception e) {}
		return null;
	}

	public int nextInt() { return Integer.parseInt(next()); }
	public double nextDouble() { return Double.parseDouble(next()); }
	public long nextLong() { return Long.parseLong(next()); }
```
Optional
This is part of kattio and can be removed.

**Comment by TruaShamu** [comment]
Optional:
On line 232,
```while (g == ans[self] || g == ans[parent]) g++;```
Can this be changed to this please?

```
while (g == ans[self] || g == ans[parent]) {
  g++;
}
```

**Review by TruaShamu — APPROVED** [review]
Other than the optional stuff above, the pr looks good to me :)

**Comment by SansPapyrus683** [comment]
sorry, my pr alr got merged so

**Pull request closed without merging**